sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -120,6 +120,11 @@ static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 	codec_mask = mach_params->codec_mask;
 	codec_count = hweight_long(codec_mask);
 
+	if (codec_count > 2) {
+		dev_warn(card->dev, "Only max two codecs supported!\n");
+		codec_count = 2;
+	}
+
```

Comment:
how would this work? You have 3 codecs, how would you select two out of three.

Reply:
@kv2019i instead of manipulating the codec count, why not change the else if down below on line 143 to handle the case count >=2 instead of count == 2?

Reply:
@plbossart Uhm, you have a point. I was assuming we enumerate the codecs in deterministic order, and we select the codec with lowest address bit first, so SOF PCM pipe is set up, the first HDA codec will be included in the pipeline. But, but, going through the code again, I'm not 100% sure the first codec is selected. Hmm, I'll see if I can guarantee this somehow. If not, I can reduce the patchset so that we only ignore probe failures.

Reply:
Ok, I think this actually does work, I double-checked it now. We hardcode the bus address in machine drivers (the "ehdaudioXDy" names), so it is guaranteed we will set up the right codec -- we always enable addr=0 as the external HDA codec, no matter if other codecs are found.

But, but, I admit this is less important. I'll split it out for now and put the multi-codec case to the enhancement list for now.